pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,90 @@
+
+Each service container can define the contracts that it enforces. If a container's contract requirements are not met, the release is not deployed to the device. You only need to define a contract for service containers that you wish to enforce a contract.
+
+## Contract specification
+
+Container contracts are defined as a `contract.yml` file and must be placed in the application structure alongside the corresponding service Dockerfile. For example, the following [multi-container application](/learn/develop/multicontainer/) defines two services named `first-service` and `second-service`, each with a corresponding `contract.yml` file.
+
+```shell
+.
+├── docker-compose.yml
+├── first-service
+│   ├── contract.yml
+│   └── Dockerfile.template
+└── second-service
+    ├── contract.yml
+    └── Dockerfile.template
+```    
+
+`contract.yml` is a YAML file containing `type`, `slug`, `name`, and `requires` keys, which define the requirements of the contract. The following example requires the device to be running a version of balenaOS greater or equal than `2.40.0` for the contract to pass.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-os-gt-240'
+name: 'Enforce hostOS Version'
+requires:
+  - type: 'sw.os'
+    version: '>=2.40.0'
+```
+
+* `type` should be set to `sw.container` for container contracts.
+* `slug` should be a unique identifier to identify the contract.
+* `name` should be a descriptive name of the contract.
+* `requires` should be a list of requirements to be enforced by the contract defined by `type` and `version` keys. Currently, `type` may be either `sw.os` for the hostOS version or `sw.supervisor` for the device Supervisor version. 
+
+__Note__ If `name` or `type` are not supplied, the build will fail. If `slug` is not provided, the build will complete, but the contract will fail validation on deployment. If `requires` is not provided, the contract will have no effect and the release will be deployed.
+
+Multiple contract requirements may be added to a single contract. The following example requires that the hostOS is greater than or equal to 2.44.0, and the Supervisor version is greater than 10.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-supervisor-and-os'
+name: 'Enforce hostOS and supervisor'
+requires:
+  - type: 'sw.supervisor'
+    version: '>10'
+  - type: 'sw.os'
+    version: '>=2.44.0'  
+```
+
+__Note__ Types in the contract requirements are not validated, so passing values for `type` other than `sw.os` or `sw.supervisor` will result in the contract passing.
+
+When deploying a release, if the contract requirements are not met, the release will fail, and all devices in the application will remain on their current release. This default behavior can be overridden by including an [optional label](#optional-contracts) in the application docker-compose.yml file. If there are any existing running services, they will not be destroyed and continue running the previous release.
```

Comment:
This reads a little weird, it's almost as if you're saying that adding the label will make the entire release ignore contracts, where as it's based on a single service.